Client/Helpers/SpotlightHelper.swift

```diff
@@ -0,0 +1,129 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+import CoreSpotlight
+import WebKit
+
+
+private let log = Logger.browserLogger
+private let browsingActivityType: String = "org.mozilla.firefox.browsing"
+
+class SpotlightHelper: NSObject {
+    private(set) var activity: NSUserActivity? {
+        willSet {
+            activity?.invalidate()
+        }
+        didSet {
+            activity?.delegate = self
+        }
+    }
+
+    private var urlForThumbnail: NSURL?
+    private var thumbnailImage: UIImage?
+
+    private let createNewTab: ((url: NSURL) -> ())?
+
+    private let profile: Profile!
+    private weak var tab: Browser?
+
+    init(browser: Browser, profile: Profile, openURL: ((url: NSURL) -> ())? = nil) {
+        createNewTab = openURL
+        self.profile = profile
+        self.tab = browser
+
+        if let path = NSBundle.mainBundle().pathForResource("SpotlightHelper", ofType: "js") {
+            if let source = try? NSString(contentsOfFile: path, encoding: NSUTF8StringEncoding) as String {
+                let userScript = WKUserScript(source: source, injectionTime: WKUserScriptInjectionTime.AtDocumentEnd, forMainFrameOnly: true)
+                browser.webView!.configuration.userContentController.addUserScript(userScript)
+            }
+        }
+    }
+
+    deinit {
+        // Invalidate the currently held user activity (in willSet)
+        // and release it.
+        self.activity = nil
+    }
+
+    func update(pageContent: [String: String], forURL url: NSURL) {
+        var activity: NSUserActivity
+        if let currentActivity = self.activity where currentActivity.webpageURL == url {
+            activity = currentActivity
+        } else {
+            activity = createUserActivity()
+            self.activity = activity
+            activity.webpageURL = url
+        }
+
+        activity.title = pageContent["title"]
+        if #available(iOS 9, *) {
```

Comment:
Is the spotlight integration an iOS 9 only feature? I wonder if we could wrap this whole helper with a iOS 9 check and conditionally add it to the tab instead of checking the condition here.